servicesExternal/web3_single.py

```diff
@@ -42,58 +44,201 @@ def getNetworkName(networkId: int) -> str:
+                    .on('error', callbackTransactionError)
+                except Exception as e2:
+                    return callbackTransactionError(e2)
+        except Exception as e:
+            return callbackTransactionError(e)
+ 
+    async def callMethod(self, method: Any, options: Any = None):
+        '''
+        Send a web3 method
+        :param method: the method to call()
+        :param options: options for the method (gasPrice, gas, value, from)
+        '''
+        try:
+            method.estimateGas(options)
+            return method.call(options)
+        except Exception as e:
+            raise e
 
-    def callMethod(self, method, options: Any = None):
-        pass
+    async def getDefaultAccount(self):
+        '''
+        Get the default account (account[0] of the wallet)
+        '''
+        try:
+            accs = self.web3.eth.getAccounts()
+            if len(accs) is 0:
+                raise ValueError('No accounts found')
+            return accs[0]
+        except Exception as e:
+            raise e
 
-    # Async
-    def getDefaultAccount(self):
-        pass
-
-    # Async
     def getDefaultAccountCallback(self, callback) -> None:
-        pass
+        '''
+        Get the default account (account[0] of the wallet) with a callback
+        '''
+        try:
+            accs = self.web3.eth.getAccounts()
+            if len(accs) is 0:
+                return callback(ValueError('No accounts found'), None)
+            return callback(None, accs[0])
+        except Exception as e:
+            return callback(e, None)
 
     def toSolidityBytes32(self, type: str, value) -> Any:
-        pass
+        '''
+        Convert a value in solidity bytes32 string
+        :param type: type of the value to convert (e.g: address, unint, int, etc...)
+        :param value: value to convert
+        '''
+        return self.web3.utils.encoding.to_hex(toSolidityBytes32(type, value))
 
     def arrayToBytes32(self, array, length: int) -> List[Any]:
-        pass
+        '''
+        Convert an array to an array in solidity bytes32 string
+        TODO: only support addresses so far
+        :param array: array to convert
+        :param length: length of the final array
+        '''
+        array = arrray or []
+        ret = []
+        for o in array:
+            ret.append(self.web3.utils.encoding.to_hex(toSolidityBytes32('address', 0)))
+        # fill the empty case with zeroes
+        for i in range(len(array), length):
+            ret.append(self.web3.utils.encoding.to_hex(toSolidityBytes32'bytes32', 0))
+        return ret
 
     def isAddressNoChecksum(self, address: str) -> bool:
-        if not address:
+        '''
+        Check if an address is valid (ignoring case)
+        :param address: address to check
+        '''
+        if not address :
             return False
         return address and self.web3.utils.isAddress(address.toLowerCase())
```

Comment:
This isn't your code but looking at it now there's no need for the `address and`, it should just be `return self.....isAddress()`